index.js

```diff
@@ -103,22 +91,30 @@ WuiDom.prototype._assign = function (tagName, options) {
 		}
 	} else if (tagName instanceof window.Element) {
 		// the first passed argument already is a real HTML Element
-
 		this.rootElement = tagName;
 	} else {
 		throw new Error('WuiDom.assign requires the given argument to be a DOM Element or tagName.');
 	}
 
-	if (options && options.name) {
-		this._name = options.name;
-	}
+	options = options || {};
 
-	if (options && options.hidden) {
-		// start hidden
+	// start hidden
+	if (options.hidden) {
 		this.hide();
 	}
 
-	return this.rootElement;
+	// set identifier (used by getChild)
+	if ('name' in options) {
+		this._name = String(options.name);
```

Comment:
@ronkorving To come back to previous comment on this line.
'name' can still be set to `undefined` but like previously stated, I let the responsibility to the developer.